lib/checkmemoryleak.cpp

```diff
@@ -1000,7 +1000,8 @@ void CheckMemoryLeakNoVar::checkForUnreleasedInputArgument(const Scope *scope)
 void CheckMemoryLeakNoVar::checkForUnusedReturnValue(const Scope *scope)
 {
     for (const Token *tok = scope->bodyStart; tok != scope->bodyEnd; tok = tok->next()) {
-        if (!Token::Match(tok, "%name% ("))
+        bool isNew = false;
+        if (!Token::Match(tok, "%name% (") && !(isNew = mTokenizer->isCPP() && tok->str() == "new"))
```

Comment:
I would prefer that the assignment is made before the condition. then isNew can be const also.